client/layout/utils.ts

```diff
@@ -140,3 +140,21 @@ export const handleScroll = ( event: React.UIEvent< HTMLElement > ): void => {
 		ticking = true;
 	}
 };
+
+export const handleScrollGlobalSidebar = ( event: React.UIEvent< HTMLElement > ): void => {
+	const globalSidebarBodyEl = document.querySelector( '.global-sidebar .sidebar__body' );
+	if ( ! globalSidebarBodyEl ) {
+		return;
+	}
+
+	if ( ! globalSidebarBodyEl.hasAttribute( 'data-scroll-initialized' ) ) {
```

Comment:
Added the event listener here since the element `.global-sidebar .sidebar__body' is not defined when `<SidebarScrollSynchronizer/>` is executed.